e2e/app/component/bypass-modules.ts

```diff
@@ -0,0 +1,54 @@
+import { Page } from 'puppeteer';
+import BaseMenu from './base-menu';
+import Checkbox from 'app/element/checkbox';
+
+const defaultXpath = '//*[@id="popup-root"]';
+
+export default class BypassPopup extends BaseMenu {
+  constructor(page: Page, xpath: string = defaultXpath) {
+    super(page, xpath);
+  }
+
+  /**
+   *  Get labels of all modules.
+   */
+  async getAllToggleTexts(): Promise<string[]> {
+    const selector = `${this.getXpath()}//label/span/text()`;
+    return this.getMenuItemTexts(selector);
+  }
+
+  /**
+   *  Get all toggle status of all modules.
+   */
+  async getAllToggleInputs(): Promise<boolean[]> {
+    const selector = `${this.getXpath()}//label/div/input[@type='checkbox']`;
+    return this.getToggleStatus(selector);
+  }
+
+  // xpath of the bypass-link-checkbox in user admin table
+  getMenuItemXpath(toggleAccessText: string): string {
+    return `${this.getXpath()}//span[text()= "${toggleAccessText}"]/preceding-sibling::div/input[@type='checkbox']`;
+  }
+
+  //get status of each bypass modules individually
+  async getEachModuleStatus(toggleAccessText: string): Promise<boolean> {
```

Comment:
Hey Joel & Alex, I've added this function which will get the status of each modules individually.

Reply:
Cool. nit: func name is confusing. How about rename to `getAccessModuleStatus(accessModule: string)`?

Reply:
sounds good!